xknx/devices/climate_mode.py

```diff
@@ -142,35 +165,20 @@ async def _set_internal_operation_mode(self, operation_mode):
 
     async def set_operation_mode(self, operation_mode):
         """Set the operation mode of a thermostat. Send new operation_mode to BUS and update internal state."""
-        if not self.supports_operation_mode:
+        if not self.supports_operation_mode or \
+                operation_mode not in self.operation_modes_:
             raise DeviceIllegalValue("operation mode not supported", operation_mode)
-        if self.group_address_operation_mode is not None:
-            await self.send(
-                self.group_address_operation_mode,
-                DPTArray(DPTHVACMode.to_knx(operation_mode)))
-        if self.group_address_operation_mode_protection is not None:
-            protection_mode = operation_mode == HVACOperationMode.FROST_PROTECTION
-            await self.send(
-                self.group_address_operation_mode_protection,
-                DPTBinary(protection_mode))
-        if self.group_address_operation_mode_night is not None:
-            night_mode = operation_mode == HVACOperationMode.NIGHT
-            await self.send(
-                self.group_address_operation_mode_night,
-                DPTBinary(night_mode))
-        if self.group_address_operation_mode_comfort is not None:
-            comfort_mode = operation_mode == HVACOperationMode.COMFORT
-            await self.send(
-                self.group_address_operation_mode_comfort,
-                DPTBinary(comfort_mode))
-        if self.group_address_controller_status is not None:
-            await self.send(
-                self.group_address_controller_status,
-                DPTArray(DPTControllerStatus.to_knx(operation_mode)))
-        if self.group_address_controller_mode is not None:
-            await self.send(
-                self.group_address_controller_mode,
-                DPTArray(DPTHVACContrMode.to_knx(operation_mode)))
+
+        for rv in self.__iter_remote_values_climate_mode():
+            if rv.writable and \
+                    operation_mode in rv.supported_operation_modes():
+                await rv.set(operation_mode)
+
+        for rv in self.__iter_remote_values_climate_binary_mode():
+            if rv.writable:
+                # foreign operation modes will set the RemoteValue to False
+                await rv.set(operation_mode)
```

Comment:
Could this lead to problems where splitted up operation_modes and eg. controller_mode is defined? Eg. controller_mode is set to "cool" should operation_mode_comfort be set to `False`? Is this constellation even a thing?
If it was we would need to check against the other 2 possible binary operation modes. This would mean to override `RemoteValue.set()` or check there against `None` from `RemoteValue.to_knx()`.

Reply:
I am not sure I get your point here. Controller and operation mode should be set independently. I can be in "cool" and still need to control if I want "comfort", "away", etc. Or am I missing your point here?

Reply:
Is it save to assume when eg. only group_address_operation_mode_night is set the supported operation modes are: Night, Standby, Comfort, Frost Protection? Even if we can not set 2 of them?

Reply:
I don't think we can know/say either way for sure. My understanding is, that a device either supports the 20.102 approach where you have one address that can handle all modes or your device splits them up into multiple 1 bit addresses, but it should then still have one address for each mode so it can still offer the same options. My Gira room controller e.g. offers both options, so they are just different ways of handling (e.g. you choose what can be better integrated into your visualization concept).
But on the other hand, there could potentially be devices that only support two modes (not sure if they would be 100% KNX conform then TBH). And even if a device supports all modes, a xknx user might still only provide the 1 bit addresses for a few of them, if they only want to use those few e.g. via HA.

Not sure if that comment helps or not 🤷 ...

Reply:
Anyway even with 1byte operation mode and controller mode it is currently not possible to set "cool" and "comfort" at the same time because a new value is always stored in `self.operation_mode`. Probably the right way would be to return a list of currently active modes or store the values separately. Then the HA integration can parse it as mode or preset - which don't really fit knx modes.